go_lib/updater/updater.go

```diff
@@ -473,40 +488,38 @@ func (u *Updater[R]) runReleaseDeploy(predictedRelease R, currentRelease *R) err
 		}
 	}
 
-	if len(u.skippedPatchesIndexes) > 0 {
-		for _, index := range u.skippedPatchesIndexes {
-			release := u.releases[index]
-			// skip not-deployed patches
-			err = u.updateStatus(release, "", PhaseSkipped)
-			if err != nil {
-				return fmt.Errorf("update status to skipped: %w", err)
-			}
-		}
-	}
-
-	return nil
+	return u.CommitSkippedReleases()
 }
 
 // PredictNextRelease runs prediction of the next release to deploy.
 // it skips patch releases and save only the latest one
-func (u *Updater[R]) PredictNextRelease() {
+func (u *Updater[R]) PredictNextRelease(release R) {
```

Comment:
what it this input for?
I think, we have to describe it in the function description

Reply:
yeah, makes sense, the reason why I decided to add an argument is because almost all "metrics" magic takes place in the updater, but at the same time I don't want to update metrics for all current releases on each `PredictNextRelease` run, so I decided that having some sort of `run-designated` release would do the trick of updating only that release's metrics.
I find it to be a product of the updater's a bit strange design :)